feat(agw): Write per subscriber APN specific data into subscriberDB store when MME receives Update Location Answer

vroon2703 opened the pull request:
<!--
    Tag your PR title with the components that it touches along with
    the type of change
    E.g. "fix(orc8r): Fix reindexer race condition" or "feat(agw) ..."
-->

## Summary
(In Progress!!)
This PR has the following new functions: 

1. SQLite functions that allow adding subscribers 
2. A function to convert ula response to subscriber data which is then written into the sqlite database

## Test Plan
Unit tests for the function that converts ula to sub_data
Test that an attach/detach is still successful 

##Note
The sqlite functions have a dependency on libsqlite3-dev 
TODO: Add it to the packaging script, but for now do `sudo apt-get install libsqlite3-dev` in the VM.

vroon2703 committed:
feat(agw): authentication done via eps_authentication instead of subscriberdb

Signed-off-by: Varun Madathil <varun27@fb.com>

vroon2703 committed:
updating flag and configurations for eps_auth

Signed-off-by: Varun Madathil <varun27@fb.com>

vroon2703 committed:
fixing typo in service_registry

Signed-off-by: Varun Madathil <varun27@fb.com>

vroon2703 committed:
removing unnecessary logs in ul

Signed-off-by: Varun Madathil <varun27@fb.com>

vroon2703 committed:
(agw) write subscriber data when MME receives ULA

Signed-off-by: vroon2703 <varun27@fb.com>

vroon2703 committed:
cleaning up formatting issues

Signed-off-by: vroon2703 <varun27@fb.com>

vroon2703 committed:
updated cmakelists

Signed-off-by: vroon2703 <varun27@fb.com>

vroon2703 committed:
resolving compile errors

Signed-off-by: vroon2703 <varun27@fb.com>

vroon2703 committed:
fixing formatting issues

Signed-off-by: vroon2703 <varun27@fb.com>

vroon2703 committed:
linking libraries to s6a_proxy

Signed-off-by: vroon2703 <varun27@fb.com>

vroon2703 committed:
updated the function to convert ula response to apn

Signed-off-by: vroon2703 <varun27@fb.com>

maxhbr commented on lte/gateway/c/core/oai/lib/store/sqlite.cpp:
```diff
@@ -0,0 +1,123 @@
+#include "lte/gateway/c/core/oai/lib/store/sqlite.hpp"
```
missing license header

vroon2703 committed:
updates to s6aclient

Signed-off-by: vroon2703 <varun27@fb.com>

vroon2703 committed:
adding unit test for converting function

Signed-off-by: vroon2703 <varun27@fb.com>

vroon2703 committed:
fixing formattting

Signed-off-by: vroon2703 <varun27@fb.com>

vroon2703 committed:
adding unit test for converting ula to subdata

Signed-off-by: vroon2703 <varun27@fb.com>

ssanadhya commented on lte/gateway/c/core/oai/lib/store/sqlite.cpp:
```diff
@@ -0,0 +1,137 @@
+ *-------------------------------------------------------------------------------
+ * For more information about the OpenAirInterface (OAI) Software Alliance:
+ *      contact@openairinterface.org
+ */
```
Please use the Magma Authors copyright here

ssanadhya commented on lte/gateway/c/core/oai/lib/store/sqlite.hpp:
```diff
@@ -0,0 +1,51 @@
+ *-------------------------------------------------------------------------------
+ * For more information about the OpenAirInterface (OAI) Software Alliance:
+ *      contact@openairinterface.org
+ */
```
Please use the Magma copyright here, since this is a new file getting added as part of Magma project.

ssanadhya commented on lte/gateway/c/core/oai/test/lib/test_ula_subData.cpp:
```diff
@@ -0,0 +1,146 @@
+#include <string.h>
```
Include Magma copyright header here too

ssanadhya commented on lte/gateway/c/core/oai/lib/s6a_proxy/S6aClient.cpp:
```diff
@@ -222,4 +222,83 @@ void S6aClient::update_location_request(
+void S6aClient::convert_ula_to_subscriber_data(
+    feg::UpdateLocationAnswer response, magma::lte::SubscriberData& sub_data) {
+  if (response.apn_size() < 1) {
+    std::cout << "No APN configurations received" << std::endl;
```
Just return from here. 

Then you can do without encapsulating the rest of the function description in else brace

ssanadhya commented on lte/gateway/c/core/oai/lib/s6a_proxy/S6aClient.cpp:
```diff
@@ -222,4 +222,83 @@ void S6aClient::update_location_request(
+                  .unit());
+        }
+        // sub_apn_config->set_allocated_ambr(
+        // (magma::lte::AggregatedMaximumBitrate*)apn.mutable_ambr());
```
Remove commented code

ssanadhya commented on lte/gateway/c/core/oai/lib/s6a_proxy/S6aClient.cpp:
```diff
@@ -222,4 +222,83 @@ void S6aClient::update_location_request(
+        if (apn.ambr().max_bandwidth_ul() != 0) {
+          ambr->set_max_bandwidth_ul(apn.ambr().max_bandwidth_ul());
+        }
+        if (apn.ambr().unit() != 0) {
```
No need for this check as value 0 indicates enum type BPS

ssanadhya commented on lte/gateway/c/core/oai/lib/s6a_proxy/S6aClient.cpp:
```diff
@@ -222,4 +222,83 @@ void S6aClient::update_location_request(
+        // (magma::lte::AggregatedMaximumBitrate*)apn.mutable_ambr());
+      }
+
+      if (apn.pdn() != 0) {
```
No need for this check as 0 indicates a valid enum type

ssanadhya commented on lte/gateway/c/core/oai/lib/s6a_proxy/S6aClient.cpp:
```diff
@@ -222,4 +222,83 @@ void S6aClient::update_location_request(
+            (magma::lte::APNConfiguration_PDNType)apn.pdn());
+      }
+
+      if (apn.served_party_ip_address_size() > 0) {
```
Add a comment that only one IP address is assigned in the current implementation

ssanadhya commented on lte/gateway/c/core/oai/lib/s6a_proxy/s6a_client_api.cpp:
```diff
@@ -144,6 +146,21 @@ static void s6a_handle_update_location_ans(const std::string& imsi,
+        sub_data.set_allocated_sid(&sub_id);
+        magma::S6aClient::convert_ula_to_subscriber_data(response, sub_data);
+        magma::lte::SqliteStore* sqlObj = new magma::lte::SqliteStore(
+            "", 2);  // hardcoded here, todo get the db location for the sqlite
```
The location of db should be same as what SubscriberDB is using currently

ssanadhya commented on lte/gateway/c/core/oai/lib/s6a_proxy/s6a_client_api.cpp:
```diff
@@ -144,6 +146,21 @@ static void s6a_handle_update_location_ans(const std::string& imsi,
+      // convert ULA response to SubscriberData and write to subscriberdb
+      if (S6aClient::get_cloud_subscriberdb_enabled()) {
+        magma::lte::SubscriberData sub_data = magma::lte::SubscriberData();
+        magma::lte::SubscriberID sub_id = magma::lte::SubscriberID();
```
No need to create a new object here. We can just use the member object in `sub_data`

ssanadhya commented on lte/gateway/c/core/oai/lib/s6a_proxy/s6a_client_api.hpp:
```diff
@@ -46,4 +46,5 @@ bool s6a_update_location_req(const s6a_update_location_req_t* const ulr_p);
 
 #ifdef __cplusplus
 }
+
```
Was this added by mistake or from formatter?

ssanadhya commented on lte/gateway/c/core/oai/lib/store/CMakeLists.txt:
```diff
@@ -0,0 +1,55 @@
+create_proto_dir("lte" LTE_OUT_DIR)
+create_proto_dir("orc8r" ORC8R_OUT_DIR)
+
+
```
Too many new lines

ssanadhya commented on lte/gateway/c/core/oai/lib/store/CMakeLists.txt:
```diff
@@ -0,0 +1,55 @@
+
+cmake_minimum_required(VERSION 3.7.2)
+
+
```
Too many newlines

ssanadhya commented on lte/gateway/c/core/oai/lib/store/CMakeLists.txt:
```diff
@@ -0,0 +1,55 @@
+
+set(SUB_CPP_PROTOS subscriberdb)
+set(DIGEST_CPP_PROTOS digest)
+set(APN_CPP_PROTOS apn)
```
```suggestion
set(SUB_LTE_PROTOS subscriberdb apn)
set(SUB_ORC8R_PROTOS digest)
```

ssanadhya commented on lte/gateway/c/core/oai/lib/store/sqlite.hpp:
```diff
@@ -0,0 +1,51 @@
+
+  // Add subscriber
+  void add_subscriber(
+      SubscriberData& subscriber_data);  // TODO: add the parameters
```
Clean up the TODO comment

ssanadhya commented on lte/gateway/c/core/oai/lib/store/sqlite.hpp:
```diff
@@ -0,0 +1,51 @@
+      SubscriberData& subscriber_data);  // TODO: add the parameters
+
+  // Delete subscriber
+  void delete_subscriber();  // TODO: add the parameters
```
```suggestion
  void delete_subscriber();  // TODO(vroon2703): add the parameters
```

ssanadhya commented on lte/gateway/c/core/oai/test/lib/CMakeLists.txt:
```diff
@@ -15,12 +15,27 @@ set(CMAKE_CXX_EXTENSIONS OFF)
+    $ENV{MAGMA_ROOT}
+    ${CMAKE_CURRENT_SOURCE_DIR}
+    $<TARGET_FILE_DIR:LIB_S6A_PROXY>
+    )
```
No need for target_include_directories here.

ssanadhya commented on lte/gateway/c/core/oai/test/lib/CMakeLists.txt:
```diff
@@ -15,12 +15,27 @@ set(CMAKE_CXX_EXTENSIONS OFF)
+
 link_directories(/usr/src/googletest/googlemock/lib/)
 
+set(MCONFIG_OUT_DIR "${CMAKE_CURRENT_BINARY_DIR}/lte/protos")
```
Why do we need to generate these protos as part of test, but not as part of the original library?

ssanadhya commented on lte/gateway/c/core/oai/lib/store/sqlite.cpp:
```diff
@@ -0,0 +1,137 @@
+}
+
+void SqliteStore::add_subscriber(SubscriberData& subscriber_data) {
+  std::string sid_s = _to_str(subscriber_data);
```
`sid_s` refers to the sid field in SubscriberData

```suggestion
  std::string sid_s = _to_str(subscriber_data.sid());
```

ssanadhya replied:
I missed the part that `_to_str` is returning the SID sub-field of SubscriberData. Resolving this conversation.

ssanadhya commented on lte/gateway/c/core/oai/lib/store/sqlite.cpp:
```diff
@@ -0,0 +1,137 @@
+  }
+}
+
+// function is hardcoded for now, will fix
```
What is hardcoded here?

ssanadhya commented on lte/gateway/c/core/oai/lib/store/sqlite.cpp:
```diff
@@ -0,0 +1,137 @@
+  if (rc2 == SQLITE_OK) {
+    sqlite3_bind_text(
+        stmt, 1, sid, 4 * strlen(sid),
+        SQLITE_STATIC);  // REVIEW THAT THE PARAMETERS HERE ARE CORRECT
```
Remove comment

ssanadhya reviewed (changes requested):
Can you add a snapshot of the newly created database entries in the test plan?

ardzoht commented on lte/gateway/c/core/oai/lib/store/sqlite.cpp:
```diff
@@ -0,0 +1,137 @@
+  const char* sql = "SELECT data FROM subscriberdb WHERE subscriber_id = ?";
+  sqlite3_stmt* stmt;
+  const char* pzTail;
+  int rc2 = sqlite3_prepare_v2(db, sql, strlen(sql), &stmt, &pzTail);
```
```suggestion
  int rc_prep = sqlite3_prepare_v2(db, sql, strlen(sql), &stmt, &pzTail);
```

ardzoht commented on lte/gateway/c/core/oai/lib/store/sqlite.cpp:
```diff
@@ -0,0 +1,137 @@
+  std::string db_location_s = _db_locations[_sid2bucket(sid)];
+  const char* db_location = db_location_s.c_str();
+  sqlite3* db;
+  int rc = sqlite3_open(db_location, &db);
```
you can just pass `db_location_s.c_str()` here instead of creating separate var

ardzoht replied:
```suggestion
  int rc_open = sqlite3_open(db_location, &db);
```

ardzoht commented on lte/gateway/c/core/oai/lib/store/sqlite.cpp:
```diff
@@ -0,0 +1,137 @@
+  }
+}
+
+void SqliteStore::add_subscriber(SubscriberData& subscriber_data) {
```
can we pass const for SubscriberData here ?

ardzoht commented on lte/gateway/c/core/oai/lib/store/sqlite.cpp:
```diff
@@ -0,0 +1,137 @@
+
+void SqliteStore::_create_store() {
+  int rc;
+  for (std::string db_location_s : _db_locations) {
```
```suggestion
  for (auto db_location_s : _db_locations) {
```

ardzoht commented on lte/gateway/c/core/oai/test/lib/test_ula_subData.cpp:
```diff
@@ -0,0 +1,146 @@
+};
+
+static magma::feg::UpdateLocationAnswer create_ula_object(
+    magma::feg::UpdateLocationAnswer &ula_object,
```
pass const reference here

vroon2703 committed:
eps_auth with streaming stopped

Signed-off-by: vroon2703 <varun27@fb.com>

vroon2703 committed:
reolving cpplint issues

Signed-off-by: vroon2703 <varun27@fb.com>

vroon2703 committed:
Update lte/gateway/c/core/oai/lib/store/sqlite.cpp

Co-authored-by: Alex Rodriguez <ardzoht@gmail.com>
Signed-off-by: vroon2703 <varun27@fb.com>

ardzoht commented on lte/cloud/go/services/subscriberdb/servicers/southbound/subscriberdb_servicer.go:
```diff
@@ -309,6 +336,23 @@ func (s *subscriberdbServicer) shouldResync(network string, gateway string) bool
+	nwCellularConfig := nwCellularConfigType.(*lte_models.NetworkCellularConfigs)
+	EpcConfig := nwCellularConfig.Epc
+	flag := EpcConfig.CloudSubscriberdbEnabled
+	return flag
```
```suggestion
	return EpcConfig.CloudSubscriberdbEnabled
```

ardzoht commented on lte/cloud/go/services/subscriberdb/servicers/southbound/subscriberdb_servicer.go:
```diff
@@ -309,6 +336,23 @@ func (s *subscriberdbServicer) shouldResync(network string, gateway string) bool
+	nwCellularConfigType, ok := network.Configs[lte.CellularNetworkConfigType]
+	if (!ok){
+		fmt.Errorf("Error fetching cellular configs")
+	}
```
Should we return a default for these cases?

ssanadhya replied:
Good point @ardzoht ! The default should be the safer approach, i.e. no subscriber data is streamed. Hence `return true` . cc: @vroon2703

vroon2703 committed:
adding s1ap test for cloud authentication

Signed-off-by: vroon2703 <varun27@fb.com>

Update lte/gateway/c/core/oai/lib/store/sqlite.cpp

Co-authored-by: Alex Rodriguez <ardzoht@gmail.com>

Update lte/gateway/c/core/oai/lib/store/sqlite.cpp

Co-authored-by: Alex Rodriguez <ardzoht@gmail.com>

rsarwad commented on lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py:
```diff
@@ -788,6 +788,19 @@ def _get_s1ap_sub(self, sid, imei):
+            subscribers.append(self._get_s1ap_sub(sid, imei))
+
+
+        self._subscriber_client.wait_for_changes()
```
Do we need "wait_for_changes()" to be called because adding subscriber through subscriber_client is commented?

rsarwad commented on lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py:
```diff
@@ -788,6 +788,19 @@ def _get_s1ap_sub(self, sid, imei):
         return ue_cfg
 
+    def add_sub_cloud(self, num_ues=1):
+        print("not adding subscriber again")
```
This fun() mainly generates IMSI and IMEI for all users as indicated in num_ues. So what's this log indicate?

rsarwad commented on lte/gateway/python/integ_tests/s1aptests/test_attach_detach_cloud.py:
```diff
@@ -0,0 +1,66 @@
+
+    def test_attach_detach_cloud(self):
+        """ Basic attach/detach test with a single UE """
+        num_ues = 1
```
Can we configure num_ues to 2, as detach_type and wait_for_s1 are used set to different values for each UE

rsarwad commented on lte/gateway/python/integ_tests/s1aptests/s1ap_wrapper.py:
```diff
@@ -368,7 +373,7 @@ def configAPN(self, imsi, apn_list, default=True, static_ip=None):
             else:
                 apn_list = [magma_default_apn]
-        self._sub_util.config_apn_data(imsi, apn_list)
+        # self._sub_util.config_apn_data(imsi, apn_list)
```
Since the invoking of configAPN is guarded with flag, cloud. Do we need to comment this line?

rsarwad commented on lte/gateway/python/integ_tests/s1aptests/s1ap_wrapper.py:
```diff
@@ -266,12 +269,14 @@ def configUEDevice(self, num_ues, req_data=None, static_ips=None):
-                default=True,
-                static_ip=ue_ip,
-            )
+            if not cloud:
```
I think similar changes would needed for fun, configUEDevice_ues_same_imsi

rsarwad commented on lte/cloud/go/services/subscriberdb/servicers/southbound/subscriberdb_servicer.go:
```diff
@@ -45,6 +45,8 @@ func NewSubscriberdbServicer(config subscriberdb.Config, store syncstore.SyncSto
 	return &subscriberdbServicer{store: store, Config: config}
 }
 
+
```
Remove extra spaces

thmsschmitt committed:
fix(ci): Use yarn for reviewdog eslint (#13458)

Signed-off-by: Thomas Schmitt <thomas.schmitt@tngtech.com>

Neudrino committed:
refactor(nms): add FEG Gateway Add Dialog Button Test (#13436)

Signed-off-by: Fritz Lehnert <13189449+Neudrino@users.noreply.github.com>

ivan-freedomfi committed:
fix(nms): ignore errors from DP when it's disabled (#13428)

Temporary hotfix for https://github.com/magma/domain-proxy/issues/469.
Ignore the error from Domain Proxy CbsdContext when Domain Proxy is
disabled. Log error to console instead of showing a notification to the
user.

Signed-off-by: Ivan Sergiienko <ivan@freedomfi.com>

ssanadhya commented on lte/cloud/go/services/subscriberdb/servicers/southbound/subscriberdb_servicer.go:
```diff
@@ -141,7 +151,16 @@ func (s *subscriberdbServicer) Sync(
+		glog.Infof("Cloud Authentication enabled, not streaming subscriber data")
+		return nil, nil
+	} else {
+		glog.Infof("Cloud Authentication enabled, not streaming subscriber data")
```
```suggestion
		glog.Infof("Cloud Authentication disabled, streaming subscriber data")
```

ssanadhya commented on lte/cloud/go/services/subscriberdb/servicers/southbound/subscriberdb_servicer.go:
```diff
@@ -204,7 +223,15 @@ func (s *subscriberdbServicer) ListSubscribers(ctx context.Context, req *lte_pro
+		glog.Infof("Cloud Authentication enabled, not streaming subscriber data")
+		return nil, nil
+	} else {
+		glog.Infof("Cloud Authentication enabled, not streaming subscriber data")
```
```suggestion
		glog.Infof("Cloud Authentication disabled, streaming subscriber data")
```

ssanadhya commented on lte/cloud/go/services/subscriberdb/servicers/southbound/subscriberdb_servicer.go:
```diff
@@ -250,6 +277,14 @@ func (s *subscriberdbServicer) getSubscribersChangeset(networkID string, clientD
+		glog.Infof("Cloud Authentication enabled, not streaming subscriber data")
+		return false , nil, nil, nil
+	} else {
+		glog.Infof("Cloud Authentication enabled, not streaming subscriber data")
```
```suggestion
		glog.Infof("Cloud Authentication disabled, streaming subscriber data")
```

ssanadhya commented on lte/cloud/go/services/subscriberdb/servicers/southbound/subscriberdb_servicer.go:
```diff
@@ -271,7 +306,16 @@ func (s *subscriberdbServicer) loadSubscribersPageFromCache(ctx context.Context,
+		glog.Infof("Cloud Authentication enabled, not streaming subscriber data")
+		return nil, "", nil
+	} else {
+		glog.Infof("Cloud Authentication NOT enabled,  streaming subscriber data")
```
```suggestion
		glog.Infof("Cloud Authentication disabled,  streaming subscriber data")
```

ssanadhya commented on lte/cloud/go/services/subscriberdb/servicers/southbound/subscriberdb_servicer.go:
```diff
@@ -309,6 +353,23 @@ func (s *subscriberdbServicer) shouldResync(network string, gateway string) bool
+	networkID := gateway.NetworkId
+	network, err := configurator.LoadNetwork(ctx, networkID, false, true,serdes.Network)
+	if err != nil {
+		fmt.Errorf("Load network for network %s: %w", networkID, err)
```
```suggestion
		fmt.Errorf("Load error for network %s: %w", networkID, err)
```

ssanadhya commented on lte/gateway/c/core/oai/lib/s6a_proxy/S6aClient.cpp:
```diff
@@ -222,4 +222,80 @@ void S6aClient::update_location_request(
+  if (response.apn_size() < 1) {
+    std::cout << "No APN configurations received" << std::endl;
+  } else {
+    std::cout << "CONVERTING ULA TO SUBSCRIBER DATA" << std::endl;
```
```suggestion
    std::cout << "Converting ULA to Subscriber Data object" << std::endl;
```

ssanadhya commented on lte/gateway/c/core/oai/lib/store/sqlite.cpp:
```diff
@@ -0,0 +1,140 @@
+  } else {
+    std::cout << "Invalid sid " << subscriber_data.sid().id() << " type "
+              << subscriber_data.sid().type() << std::endl;
+  }
```
This should still return some value.

ssanadhya commented on lte/gateway/c/core/oai/lib/store/sqlite.cpp:
```diff
@@ -0,0 +1,140 @@
+
+void SqliteStore::add_subscriber(const SubscriberData& subscriber_data) {
+  std::string sid_s = _to_str(subscriber_data);
+  const char* sid = sid_s.c_str();
```
`sid_s` is only being used once. Can we modify `_to_str` to return a C string instead of std::string.

ssanadhya reviewed (changes requested):
@vroon2703 , it will be good to split the S1ap integration tests into a separate PR . This PR should focus on writing APN info in subscriberDB , along with unit tests and changes in Ansible files to install sqlite library in gateway VM. This will make it easier to review and merge

Sathyaj27 committed:
fix(amf): Reverting Guti creation during reg-accept (#13454)

Signed-off-by: Sathyaj27 <sathya.jayadev@wavelabs.ai>

vroon2703 committed:
merge conflcts

Signed-off-by: vroon2703 <varun27@fb.com>

vroon2703 committed:
reverting s1ap test files

Signed-off-by: vroon2703 <varun27@fb.com>

vroon2703 committed:
deleting file

Signed-off-by: vroon2703 <varun27@fb.com>

vroon2703 committed:
adding sqlite dependency

Signed-off-by: vroon2703 <varun27@fb.com>

jkmar committed:
feat(dp): Support fetching multiple grants for given cbsd (#13453)

* feat(dp): Support fetching multiple grants for given cbsd

Signed-off-by: Kuba Marciniszyn <kuba@freedomfi.com>

* feat(nms): support multiple grants per CBSD

- Add new column "Grant Status" to CBSD table
- Update NMS CBSDs Table to handle multiple grants per CBSD
  (previously we had only one).
  This means each CBSD can have multiple values for each of
  the following fields:
  - bandwidth_mhz
  - frequency_mhz
  - grant_expire_time
  - max_eirp
  - status
  - transmit_expire_time

For each of the fields the table now displays multiple values in a
single cell, with <hr /> seprator.

Note: this is a quick solution to get it into v1.8 release.
It is planned to improve UX after the release.

Signed-off-by: Ivan Sergiienko <ivan@freedomfi.com>

Co-authored-by: Ivan Sergiienko <ivan@freedomfi.com>

Commit:
chore(dp): do not create idle grants (#13465)

* chore(dp): do not create idle grants

Signed-off-by: Jarosław Jaszczuk <jaroslaw@freedomfi.com>

* CR fixes
* downgrade
* migration unit tests

Signed-off-by: Jarosław Jaszczuk <jaroslaw@freedomfi.com>

Akshayp77 committed:
fix(amf): Service-accept for request without uplink data status (#13031)

Signed-off-by: Akshayp77 <aksahy.patidar@wavelabs.ai>

Co-authored-by: Akshayp77 <aksahy.patidar@wavelabs.ai>

vroon2703 committed:
addressing review comments

Signed-off-by: vroon2703 <varun27@fb.com>

vroon2703 committed:
formatting cpp files

Signed-off-by: vroon2703 <varun27@fb.com>

vroon2703 committed:
updating with cmake and bazel files

Signed-off-by: vroon2703 <varun27@fb.com>

vroon2703 committed:
removing changes to ul

Signed-off-by: vroon2703 <varun27@fb.com>
